src/components/ShoppingListItem/ShoppingListItem.js

```diff
@@ -1,5 +1,42 @@
-const ShoppingListItem = ({ item }) => {
-  return <li className="shopping-list__item item">{item.itemName}</li>;
+import { useState, useEffect } from 'react';
+
+import isUnder24hSincePurchased from '../../utils/isUnder24hSincePurchased.js';
+
+const ShoppingListItem = ({ listId, itemId, item, handleCheck }) => {
+  const [recentlyPurchased, setRecentlyPurchased] = useState(false);
+
+  // update whether item is recently purchased
+  useEffect(() => {
+    // make sure properties exist and are not null
+    if (
+      'lastPurchaseDate' in item &&
+      item.lastPurchaseDate &&
+      'seconds' in item.lastPurchaseDate
+    )
+      setRecentlyPurchased(
+        isUnder24hSincePurchased(item.lastPurchaseDate.seconds),
```

Comment:
Is there a reason for using .seconds here?

Reply:
What actually gets returned is a Firebase object (try out logging it to the console and you'll see what I mean). To get the actual seconds (as in Unix epoch seconds if I'm understanding correctly), we needed to use the seconds property.